Bottom: 5ec6d3ef1756ed0ece94a823e935dc021fe2be06
Top:    3783139efedab3ea13b321ae215ddbba2c8980e0
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-01-22 17:49:12 +0000

Refresh of other-drawing-api-changes

---

diff --git a/src/draw.c b/src/draw.c
index fc47717..fde150f 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -1798,10 +1798,15 @@ draw_api_new (void)
   dapi->set_color_for_pin     = set_color_for_pin;
   dapi->set_color_for_via     = set_color_for_via;
   dapi->set_color_for_pad     = set_color_for_pad;
-  dapi->set_color_for_line    = set_color_for_line;
-  dapi->set_color_for_arc     = set_color_for_arc;
-  dapi->set_color_for_text    = set_color_for_text;
-  dapi->set_color_for_polygon = set_color_for_polygon;
+
+  /* The following types can be on copper layer, silk layers, selected, found,
+   * warned, they can be drawn as overlay text. We sould consisder whether
+   * any of these need to be split up into multiple hooks.
+   */
+  dapi->set_color_for_line    = NULL; // set_color_for_line;
+  dapi->set_color_for_arc     = NULL; // set_color_for_arc;
+  dapi->set_color_for_text    = NULL; // set_color_for_text;
+  dapi->set_color_for_polygon = NULL; // set_color_for_polygon;
 #if 0
   dapi->draw_rat              =
 #endif
diff --git a/src/draw.h b/src/draw.h
index 2697b0d..8ad0a78 100644
--- a/src/draw.h
+++ b/src/draw.h
@@ -47,7 +47,6 @@ void DrawPadName (PadType *);
 void DrawLine (LayerType *, LineType *);
 void DrawArc (LayerType *, ArcType *);
 void DrawText (LayerType *, TextType *);
-void DrawTextLowLevel (DrawAPI *dapi, TextType *, Coord);
 void DrawPolygon (LayerType *, PolygonType *);
 void DrawElement (ElementType *);
 void DrawElementName (ElementType *);
diff --git a/src/print.c b/src/print.c
index 85a0ea6..25956ad 100644
--- a/src/print.c
+++ b/src/print.c
@@ -98,7 +98,7 @@ text_at (DrawAPI *dapi, int x, int y, int align, char *fmt, ...)
   t.X -= w * (align & 3) / 2;
   if (t.X < 0)
     t.X = 0;
-  DrawTextLowLevel (dapi, &t, 0);
+  dapi->draw_pcb_text (dapi, NULL, &t, 0);
   if (align & 8)
     dapi->gapi->draw_line (dapi->gc, t.X,     t.Y + SCALE_TEXT (font->MaxHeight, t.Scale) + MIL_TO_COORD(10),
                                      t.X + w, t.Y + SCALE_TEXT (font->MaxHeight, t.Scale) + MIL_TO_COORD(10));
@@ -318,18 +318,20 @@ PrintFab (DrawAPI *dapi)
       dapi->gapi->set_line_width (dapi->gc, MIL_TO_COORD(10));
       LINE_LOOP (layer);
       {
+        // dapi->draw_pcb_line (dapi, layer, line);
         dapi->gapi->draw_line (dapi->gc, line->Point1.X, line->Point1.Y, line->Point2.X, line->Point2.Y);
       }
       END_LOOP;
       ARC_LOOP (layer);
       {
+        // dapi->draw_pcb_arc (dapi, layer, arc);
         dapi->gapi->draw_arc (dapi->gc, arc->X, arc->Y,
                               arc->Width, arc->Height, arc->StartAngle, arc->Delta);
       }
       END_LOOP;
       TEXT_LOOP (layer);
       {
-	DrawTextLowLevel (dapi, text, 0);
+        dapi->draw_pcb_text (dapi, layer, text, 0);
       }
       END_LOOP;
       dapi->gapi->set_line_width (dapi->gc, FAB_LINE_W);
